lib/web_ui/lib/src/engine/util.dart

```diff
@@ -678,16 +678,21 @@ void setClipPath(DomElement element, String? value) {
   }
 }
 
-void setThemeColor(ui.Color color) {
+void setThemeColor(ui.Color? color) {
   DomHTMLMetaElement? theme =
       domDocument.querySelector('#flutterweb-theme') as DomHTMLMetaElement?;
-  if (theme == null) {
-    theme = createDomHTMLMetaElement()
-      ..id = 'flutterweb-theme'
-      ..name = 'theme-color';
-    domDocument.head!.append(theme);
+
+  if (color != null) {
+    if (theme == null) {
+      theme = createDomHTMLMetaElement()
+        ..id = 'flutterweb-theme'
+        ..name = 'theme-color';
+      domDocument.head!.append(theme);
+    }
+    theme.content = colorToCssString(color)!;
+  } else if (theme != null) {
+    theme.remove();
   }
```

Comment:
No need to check `if (theme != null)`, you can just remove it if it's not null with `?.`:

```dart
if (color != null) {
  ...
} else {
  theme?.remove();
}
```